ol/smoke-tests/src/tests.rs

```diff
@@ -1,15 +1,236 @@
+    op_tool: diem_operational_tool::test_helper::OperationalTool,
+    backend: diem_config::config::SecureBackend,
+    storage: diem_secure_storage::Storage,
+    env: String,
+}
+
+impl TestEnvionement {
+    async fn new() -> TestEnvionement {
+        let env = if std::env::var("NODE_ENV") == Ok("test".to_string()) { "test" } else { "stage" };
+        let metadata = metadata().unwrap();
+        let (
+            swarm,
+            op_tool,
+            backend,
+            storage
+        ) = launch_swarm_with_op_tool_and_backend(1).await;
+
+        let client = swarm.validators().next().unwrap().rest_client();
+
+        let owner_account = storage.get::<AccountAddress>(OWNER_ACCOUNT).unwrap().value;
+        let keys = storage.export_private_key(OWNER_KEY).unwrap();
+        let local_acct = LocalAccount::new(owner_account, keys, 0);
+
+        TestEnvionement {
+            metadata,
+            swarm,
+            op_tool,
+            backend,
+            storage,
+            env: env.to_string(),
+        }
+    }
+
+    fn get_persona_mnemonic(&self, persona: &str) -> String {
+        let mnemonic_path = self.metadata.workspace_root.join(
+            format!(
+                "ol/fixtures/mnemonic/{}.mnem",
+                persona
+            )
+        );
+        std::fs::read_to_string(mnemonic_path).unwrap()
+    }
+
+    fn get_persona_wallet(&self, persona: &str) -> WalletLibrary {
+        let mnemonic_path = self.metadata.workspace_root.join(
+            format!(
+                "ol/fixtures/mnemonic/{}.mnem",
+                persona
+            )
+        );
+        let mnemonic = self.get_persona_mnemonic(persona);
+        let mnem = Mnemonic::from(&mnemonic).unwrap();
+
+        let mut wallet = WalletLibrary::new_from_mnemonic(mnem);
+        wallet.generate_addresses(1).unwrap();
+
+        wallet
+    }
+
+    async fn fund_account(&mut self, recipient: &LocalAccount) {
+        let owner_account = self.storage
+            .get::<AccountAddress>(OWNER_ACCOUNT)
+            .unwrap()
+            .value;
+
+        let keys = self.storage.export_private_key(OWNER_KEY).unwrap();
+        let sending_account = LocalAccount::new(
+            owner_account,
+            keys,
+            0,
+        );
+
+        self.swarm.chain_info().ol_create_account_by_coin(sending_account, &recipient).await.unwrap();
+    }
+
+    fn get_bin(&self, name: &str) -> Result<PathBuf> {
+        let target_directory = &self.metadata.target_directory;
+        let bin_path = target_directory.join(
+            format!(
+                "debug/{}{}",
+                name,
+                env::consts::EXE_SUFFIX
+            )
+        );
+        if !bin_path.exists() {
+            bail!(
+                "Can't find binary {} at expected path {:?}",
+                name,
+                bin_path
+            );
+        }
+        Ok(bin_path)
+    }
+
+    fn cargo_build(&mut self, package: &str) -> Result<()> {
```

Comment:
For these kinds of things, using Command. You can just run in rust debug mode:

`.args["cargo", "run", "-p", "tower"] ` etc..

Then you don't need to pre-build